src/apps/watchfaces/OswAppWatchfaceMonotimer.cpp

```diff
@@ -0,0 +1,146 @@
+        gfx->drawTick(cx, cy, r1, r2, (i * deltaAngle) + shift, color);
+    }
+}
+
+void OswAppWatchfaceMonotimer::drawNShiftedMaskedTicks(Graphics2D *gfx, uint8_t cx, uint8_t cy, uint8_t r1, uint8_t r2, uint8_t nTicks, float shift, uint16_t m, uint16_t color) {
+    const float deltaAngle = 360.0 / nTicks;
+    for(uint16_t i=0; i<nTicks; ++i) {
+        if(i%m) {
+            gfx->drawTick(cx, cy, r1, r2, (i * deltaAngle) + shift, color);
+        }
+    }
+}
+
+void OswAppWatchfaceMonotimer::drawWatch() {
+    OswHal* hal = OswHal::getInstance();
+
+    // hours
+    hal->gfx()->drawNTicks(120, 120, 117, 100, 12, ui->getForegroundColor());
+    // 30 minutes
+    drawNShiftedTicks(hal->gfx(), 120, 120, 117, 105, 12, 360.0/24.0, ui->getForegroundColor());
+    // 15 minutes
+    drawNShiftedTicks(hal->gfx(), 120, 120, 110, 105, 24, 360.0/48.0, ui->getForegroundColor());
+    // 5 minutes
+    drawNShiftedMaskedTicks(hal->gfx(), 120, 120, 109, 108, 144, 0.0, 3, ui->getForegroundColor());
+
+    // hour labels
+    hal->gfx()->setTextSize(2);
+    hal->gfx()->setTextMiddleAligned();
+
+    static const uint8_t positions[]= {
+        155, 45, 	// 01
+        180, 75, 	// 02
+        200, 120,	// 03
+        183, 165, 	// 04
+        155, 197, 	// 05
+        110, 210, 	// 06
+        65, 197, 	// 07
+        37, 165, 	// 08
+        23, 120, 	// 09
+        37, 75, 	// 10
+        65, 45, 	// 11
+        110, 30 	// 12
+    };
+
+    for(uint8_t i=0; i<12; ++i) {
+        hal->gfx()->setTextCursor(positions[i<<1], positions[(i<<1)|1]);
+        hal->gfx()->printDecimal(i+1, 2);
+    }
+
+#if OSW_PLATFORM_ENVIRONMENT_ACCELEROMETER == 1
+    const uint32_t steps = hal->environment->getStepsToday();
+    const uint32_t stepsTarget = OswConfigAllKeys::stepsPerDay.get();
+
+    hal->gfx()->setTextCenterAligned();
+    hal->gfx()->setTextSize(1);
+    hal->gfx()->setTextCursor(120, 135);
+    hal->gfx()->setTextColor(steps>stepsTarget?ui->getSuccessColor():ui->getInfoColor());
+
+    char buf[16];
+    snprintf(buf, 16, "%u", steps);
+    hal->gfx()->print(buf);
+#endif
+
+    // ticks
+    uint32_t second = 0;
+    uint32_t minute = 0;
+    uint32_t hour = 0;
+    hal->getLocalTime(&hour, &minute, &second);
+
+    if(OswConfigAllKeys::settingDisplayDualHourTick.get()) {
+        uint32_t dualSecond = 0;
+        uint32_t dualMinute = 0;
+        uint32_t dualHour = 0;
+        hal->getDualTime(&dualHour, &dualMinute, &dualSecond);
+
+        hal->gfx()->drawThickTick(120, 120, 0, 105, (360.0 * (60 * dualHour + dualMinute)) / 720.0, 1, ui->getBackgroundDimmedColor());
+    }
+
+    hal->gfx()->drawThickTick(120, 120, 0, 105, (360.0 * (60 * hour + minute)) / 720.0, 1, ui->getForegroundColor());
+
+    hal->gfx()->fillEllipse(120, 120, 4, 4, ui->getForegroundColor());
+}
+
+#ifdef GIF_BG
+OswAppGifPlayer* bgGif = new OswAppGifPlayer();
+#endif
+
+void OswAppWatchfaceMonotimer::setup() {
+#ifdef GIF_BG
+    bgGif->setup(hal);
+#endif
+#ifdef ANIMATION
```

Comment:
Can you remove this part? This part will be refactored later.

Reply:
Do you mean the bgGif- or the ANIMATION-stuff?

Reply:
ANIMATION-stuff.

Reply:
Ok.